### PR TITLE
add unit tests, package.json, CI integration

### PR DESCRIPTION
**Pull request by emrekeskinmac**

### CODE REVIEW

File: src/test/BooleanFilter.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import * as test from 'tape';
+import * as parser from 'query-string-parser';
+import { BooleanFilter, BooleanFilterOperation } from '..';
```
Comment (aacanakin) [inline src/test/BooleanFilter.ts]: Can you modify this import to the following for faster test loading?
```ts
import { BooleanFilter, BooleanFilterOperation } from '../BooleanFilter';
```

File: src/test/DateFilter.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import * as test from 'tape';
+import * as parser from 'query-string-parser';
+import * as Moment from "moment";
+import { DateFilter, DateFilterOperation } from '..';
```
Comment (aacanakin) [inline src/test/DateFilter.ts]: Can you modify this import to the following for faster test loading?
```ts
import { DateFilter, DateFilterOperation } from '../DateFilter';
```

File: src/test/NumericFilter.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import * as test from 'tape';
+import * as parser from 'query-string-parser';
+import { NumericFilter, NumericFilterOperation } from '..';
```
Comment (aacanakin) [inline src/test/NumericFilter.ts]: Can you modify this import to the following for faster test loading?
```ts
import { NumericFilter, NumericFilterOperation } from '../NumericFilter';
```

File: src/test/Pagination.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import * as test from 'tape';
+import * as parser from 'query-string-parser';
+import { Pagination } from '..';
```
Comment (aacanakin) [inline src/test/Pagination.ts]: Can you modify this import to the following for faster test loading?
```ts
import { Pagination } from '../Pagination';
```

File: src/test/StringBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import * as test from 'tape';
+import * as parser from 'query-string-parser';
+import { StringFilter, StringFilterOperation } from '..';
```
Comment (aacanakin) [inline src/test/StringBuilder.ts]: Can you modify this import to the following for faster test loading?
```ts
import { StringFilter, StringFilterOperation } from '../StringFilter';
```

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    "declaration": true,
+    "noImplicitAny": false,
+    "allowSyntheticDefaultImports": false,
+    "pretty": true,
+    "removeComments": true,
+    "lib": ["es6", "esnext.asynciterable"],
+    "typeRoots": ["node_modules/@types"],
+    "types": ["@types/node"],
+    "rootDir": "./src",
+    "outDir": "./lib"
```
Comment (aacanakin) [inline tsconfig.json]: For conventions, can you change `"outDir": "./dist"`?

File: README.md [hunk 7]
```diff
@@ -1,5 +1,9 @@
 # Dynamic Query Builder Client
 
+[![CircleCI](https://circleci.com/gh/emrekeskinmac/dynamic-query-builder-client/tree/feature%2Funit-tests.svg?style=svg)](https://circleci.com/gh/emrekeskinmac/dynamic-query-builder-client/tree/feature%2Funit-tests)
```
Comment (aacanakin) [inline README.md]: I'll provide here oplog account for circle ci as soon as possible

**Review by aacanakin — COMMENTED** [review]
Hi @emrekeskinmac,
Thanks for your great contribution.
Currently, we are using jest internally but that's ok for now because there were no open source tests before your contribution.
I have added small change requests for your pull request.
I'll provide circle ci account here as soon as possible. Ping @ardadevelioglu

**Comment by emrekeskinmac** [comment]
Hi @aacanakin 
Thank you for accepting PR. Great progress for me. 
I'm editing my mistakes and waiting for CircleCI account.

Thank you so much @aacanakin  @ardadevelioglu

File: README.md (hunk 7 above)
Comment (aacanakin) [reply README.md]: We have created circle ci account for oplog. Can you change status badge to the following;
```
[![CircleCI](https://circleci.com/gh/oplog/dynamic-query-builder-client.svg?style=svg)](https://circleci.com/gh/oplog/dynamic-query-builder-client)
```

**Review by aacanakin — CHANGES_REQUESTED** [review]

**Comment by emrekeskinmac** [comment]
@aacanakin I have edited CircleCI status badge.

**Comment by aacanakin** [comment]
@emrekeskinmac Thanks

**Review by aacanakin — APPROVED** [review]

**Pull request merged**